### ISSUE TITLE
rlwrap 43 with Node REPL is not clearing the prompt after the first <Ctrl+c> like native REPL does, even with rlwrap-direct-keypress

### ISSUE THREAD

**Issue by benintech** [hanslub42/rlwrap#101]
Hello,

Is there a way to use <Ctrl+c> to clear the current prompt, like in native REPL ? Right now I have to use backspaces to clear the prompt when using rlwrap.

Natively, if I send <Ctrl+c> in a native REPL session while typing the beginning of a command, it :
- clears the prompt (on first ctrl+c)
- then prints "(To exit, press ^C again or type .exit)" (on second ctrl+c)
- the exits REPL (on third ctrl+c)

Example : 
`$ node`
`> something   (sending first ctrl+c)`
`>   (prompt is cleared, sending second ctrl+c)` 
`(To exit, press ^C again or type .exit)`
`>    (sending third ctrl+c)`
`$    (back to shell prompt, after THREE ctrl+c's)`

I'd like to have the same behaviour with rlwrap, but it seems to be skipping the first ctrl+c.
Here's what I did : 
- I made a basic .js file to define REPL's options, let's say myrepl.js : 
`#!/usr/bin/env node`
`var options = { ignoreUndefined: true }`
`var myrepl = require('repl').start(options);`
- Then I modified .inputrc, to pass ctrl+c directly to node, bypassing rlwrap :
`$if myrepl.js`
`   "\C-c": rlwrap-direct-keypress`
`$endif`
- Finally, following Hans Lub's answer [here](https://stackoverflow.com/questions/21863836/using-rlwrap-with-node-js-repl-how-could-node-break-ctrl-c-not-be-interpre#answer-21975508), I launched rlwrap with the appropriate options :
`$ rlwrap --polling --always-readline myrepl.js`
`> something   (sending first ctrl+c)`
`>`
`(To exit, press ^C again or type .exit)`
`> something    (sending second ctrl+c, prompt is NOT cleared and still displays "something")`
`$    (back to shell prompt, after only TWO ctrl+c's)`
The prompt is NOT cleared (still displaying "something" after the first ctrl+c), and the first ctrl+c is interpreted like the second one is with native REPL.
So, basically this means you cannot use ctrl+c to clear the prompt after having started to type a command (you have to clear it with backspaces).

Any way around this ? Is there a way to make it behave like native REPL ?

**Comment by SamSavitz** [comment]
In my application, wrapping [Magma](http://magma.maths.usyd.edu.au/magma/), I also find that Ctrl-C fails to clear the line, but Ctrl-C + Enter has the desired behavior.  It's not a complete solution, but it's workable.

**Comment by hanslub42** [comment]
This should be solvable by issuing a `stty intr undef` before launching `rlwrap`. Please reopen if not.
